viewer/rust/lib.rs

```diff
@@ -17,27 +17,11 @@ impl CtmResult {
         Uint32Array::from(self.file.indices.as_slice())
     }
     pub fn vertices(&self) -> Float32Array {
-        let data_as_vector3 = &self.file.vertices;
-        let vertices_as_f32 = unsafe {
-            std::slice::from_raw_parts(
-                data_as_vector3.as_ptr() as *const f32,
-                data_as_vector3.len() * 3,
-            )
-        };
-        Float32Array::from(vertices_as_f32)
+        Float32Array::from(self.file.vertex_components.as_slice())
     }
     pub fn normals(&self) -> Option<Float32Array> {
-        let data_as_vector3 = match &self.file.normals {
-            Some(x) => x,
-            None => return None,
-        };
-        let data_as_f32 = unsafe {
-            std::slice::from_raw_parts(
-                data_as_vector3.as_ptr() as *const f32,
-                data_as_vector3.len() * 3,
-            )
-        };
-        Some(Float32Array::from(data_as_f32))
+        let components = self.file.normal_components.as_ref()?;
+        Some(Float32Array::from(components.as_slice()))
```

Comment:
This could also be done with a map
`self.file.normal_components.map(|x| Float32Array::from(x.as_slice()))`
but idk if that's more clear or not